src/Repl/Decode.tsx

```diff
@@ -0,0 +1,200 @@
+  // map over every row in the board, and for every space decode the value
+  // join elms in rows by tabs, and join rows by line breaks
+  // double line break at front to differentiate between repeated boards in tuples
+  return "\n\n"+board.map(
+    (row) => row.map(
+      (space) => decodeValue(space[1])
+    ).join("\t")
+  ).join("\n");
+}
+
+
+/*
+* decodeTuple
+* Takes an JS array representing a tuple, which
+* can be nested repeatedly
+*
+* tuple: Array representing an encoded tuple
+*/
+const decodeTuple = (tuple: any) => {
+  // map a decode across all values, and join w/ ','s
+  return "(" + tuple.map((e) => decodeValue(e)).join(",") + ")";
+}
+
+
+/*
+ * decodeBool
+ * Decodes a Bool
+ *
+ * decoded : Decoded response that contains a bool
+ */
+const decodeBool = (decoded) => {
+  return decoded ? "True" : "False";
+}
+
+
+/*
+ * decodeExprType
+ * Decodes the type of the given expression
+ *
+ * latest : JSON obj of the latest expression
+ */
+const decodeExprType = (typ) => {
+  // report this type, if marked with ':t ' from the cmd line
+  //let typ = latest["contents"][1]["type"];
+  // get first letter in lowercase
+  let fl = typ[0].toLowerCase();
+  let res = " is";
+
+  if(fl === 'a' || fl === 'e' || fl === 'i' || fl === 'o' || fl === 'u') {
+    // vowel sub y, use 'an'
+    res += " an ";
+
+  } else {
+    // consonant
+    res += " a ";
+
+  }
+  return res + typ;
+}
+
+
+/*
+ * decodeError
+ * Decodes an error related to a BoGL API request
+ *
+ * error :      Obj of error that has occurred
+ * respStatus : Status code of the response
+ */
+const decodeError = (error, respStatus : number) => {
+
+  console.error(error)
+
+  if((error instanceof SyntaxError || (error.name && error.name === "SyntaxError")) && respStatus === 504) {
+    // gateway timeout
+    return " 🤖 BoGL Says: Unable to finish running your program, or not currently online. Double check your code, or check back later! ";
+
+  } else if((error instanceof SyntaxError || (error.name && error.name === "SyntaxError"))) {
+    // bad parse error
+    return " 🤖 BoGL Says: I couldn't understand your program. Please double check it and try again! ";
+
+  } else if((error instanceof TypeError || (error.name && error.name === "TypeError")) && respStatus === 0) {
+    // likely JS disabled
+    return " 🤖 BoGL Says: Unable to execute your program. Make sure that Javascript is enabled and try again! ";
+
+  } else if((error instanceof TypeError || (error.name && error.name === "TypeError"))) {
+    // something else?
+    return " 🤖 BoGL Says: Unable to execute your program, please double check your code and try again. ";
+
+  } else {
+    // general error
+    return " 🤖 BoGL Says: An error occurred: " + error + " ";
+
+  }
+}
+
+
+/*
+ * decodeReponse
+ * Decodes a response from the BoGL API, returned a decoded object
+ *
+ * reponses: JSON obj containing the responses returned
+ */
+const decodeResponse = (responses: any) => {
```

Comment:
I can add additional comments and clean this part up a bit more. I didn't like how it turned out.